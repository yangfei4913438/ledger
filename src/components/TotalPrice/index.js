import React, { Fragment } from 'react';
import { jsonLang } from '../../lang'

const TotalPrice = (props) => {
  const { total_income, total_expense } = props;
  return (
    <Fragment>
      {jsonLang.label.income}: <span id='income'>{total_income}</span> {jsonLang.label.yuan}
      <span style={{marginRight: '20px'}}/>
      {jsonLang.label.expense}: <span id='expense'>{total_expense}</span> {jsonLang.label.yuan}
    </Fragment>
  )
};

export default TotalPrice;
