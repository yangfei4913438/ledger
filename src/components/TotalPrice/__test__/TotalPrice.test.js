import React from 'react';
import { shallow } from 'enzyme';
import TotalPrice from '../index'

const props = {
  total_income: 1000,
  total_expense: 800
};

let wrapper;
describe('测试价格汇总组件', () => {
  // 每次测试前都会执行
  beforeEach(() => {
    wrapper =  shallow(<TotalPrice {...props} />);
  });
  it('匹配组件快照', () => {
    expect(wrapper).toMatchSnapshot()
  });
  it('测试收入金额和支出金额', () => {
    expect(wrapper.find('#income').text() * 1).toEqual(1000);
    expect(wrapper.find('#expense').text() * 1).toEqual(800);
  })
});
