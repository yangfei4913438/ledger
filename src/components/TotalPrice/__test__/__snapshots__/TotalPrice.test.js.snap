// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试价格汇总组件 匹配组件快照 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalPrice
    total_expense={800}
    total_income={1000}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "Income",
        ": ",
        <span
          id="income"
        >
          1000
        </span>,
        " ",
        "Yuan",
        <span
          style={
            Object {
              "marginRight": "20px",
            }
          }
        />,
        "Expense",
        ": ",
        <span
          id="expense"
        >
          800
        </span>,
        " ",
        "Yuan",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Income",
      ": ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1000,
          "id": "income",
        },
        "ref": null,
        "rendered": 1000,
        "type": "span",
      },
      " ",
      "Yuan",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "style": Object {
            "marginRight": "20px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "Expense",
      ": ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 800,
          "id": "expense",
        },
        "ref": null,
        "rendered": 800,
        "type": "span",
      },
      " ",
      "Yuan",
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "Income",
          ": ",
          <span
            id="income"
          >
            1000
          </span>,
          " ",
          "Yuan",
          <span
            style={
              Object {
                "marginRight": "20px",
              }
            }
          />,
          "Expense",
          ": ",
          <span
            id="expense"
          >
            800
          </span>,
          " ",
          "Yuan",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Income",
        ": ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1000,
            "id": "income",
          },
          "ref": null,
          "rendered": 1000,
          "type": "span",
        },
        " ",
        "Yuan",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "style": Object {
              "marginRight": "20px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "Expense",
        ": ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 800,
            "id": "expense",
          },
          "ref": null,
          "rendered": 800,
          "type": "span",
        },
        " ",
        "Yuan",
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
